enterprise/internal/campaigns/service/service.go

```diff
@@ -15,21 +15,20 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/campaigns"
 	"github.com/sourcegraph/sourcegraph/internal/db"
 	"github.com/sourcegraph/sourcegraph/internal/extsvc/auth"
-	"github.com/sourcegraph/sourcegraph/internal/httpcli"
 	"github.com/sourcegraph/sourcegraph/internal/repos"
 	"github.com/sourcegraph/sourcegraph/internal/repoupdater"
 	"github.com/sourcegraph/sourcegraph/internal/trace"
 )
 
 // New returns a Service.
-func New(store *store.Store, cf *httpcli.Factory) *Service {
-	return NewWithClock(store, cf, store.Clock())
+func New(store *store.Store) *Service {
+	return NewWithClock(store, store.Clock())
 }
 
 // NewWithClock returns a Service the given clock used
 // to generate timestamps.
-func NewWithClock(store *store.Store, cf *httpcli.Factory, clock func() time.Time) *Service {
-	svc := &Service{store: store, sourcer: repos.NewSourcer(cf), clock: clock}
+func NewWithClock(store *store.Store, clock func() time.Time) *Service {
+	svc := &Service{store: store, sourcer: repos.NewSourcer(nil), clock: clock}
```

Comment:
I think this is better since it's more explicit.

```suggestion
	svc := &Service{store: store, sourcer: repos.NewSourcer(httpcli.NewExternalHTTPClientFactory()), clock: clock}
```